transmart-core-db/grails-app/services/org/transmartproject/db/clinical/CrossTableService.groovy

```diff
@@ -0,0 +1,66 @@
+package org.transmartproject.db.clinical
+
+import groovy.transform.CompileStatic
+import org.springframework.beans.factory.annotation.Autowired
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.multidimquery.CrossTable
+import org.transmartproject.core.multidimquery.CrossTableResource
+import org.transmartproject.core.multidimquery.PatientSetResource
+import org.transmartproject.core.multidimquery.query.Combination
+import org.transmartproject.core.multidimquery.query.Constraint
+import org.transmartproject.core.multidimquery.query.Operator
+import org.transmartproject.core.multidimquery.query.PatientSetConstraint
+import org.transmartproject.core.users.User
+import org.transmartproject.db.multidimquery.CrossTableImpl
+
+@CompileStatic
+class CrossTableService extends AbstractDataResourceService implements CrossTableResource {
+
+    @Autowired
+    PatientSetResource patientSetResource
+
+    @Autowired
+    AggregateDataService aggregateDataService
+
+    @Override
+    CrossTable retrieveCrossTable(List<Constraint> rowConstraints, List<Constraint> columnConstraints,
+                                           Long patientSetId, User user) {
+
+        Constraint patientSetConstraint = new PatientSetConstraint(patientSetId)
```

Comment:
Nice, this suggests that we could easily switch to passing arbitrary constraints instead of a patient set id. Which solution do you think is better? I initially thought passing a patient set id was a good idea, but it makes the cross table functionality less flexible.

Reply:
I also saw that supporting only patient set id may cause some problems (described in TMT-219). Would this be a good solution to support both or would it be too messy? Between id and  constraint I would rather change it to constraint. I agree it would be more flexible (it could be combined with `GET /v2/patient_sets/{resultInstanceId}` to get patient set constraint based on id) and would solve the mentioned issues.